src/ultrasonic/main.c

```diff
@@ -1,13 +1,56 @@
 /**
  * main (entrypoint) implementation
  */
+#include "timercontrol.h"
 #include "init.h"
+#include "pinmap.h"
+#include <avr/io.h>
+#include <util/delay.h>
+#include <avr/interrupt.h>
+#include "stdint.h"
+#include "bit_utils.h"
+#include "globals.h"
+
+
+float getDistance(int trigPin, int frequency, uint8_t flag) {
+	PORTB.OUT &= ~(1<<trigPin); // Ensure clean HIGH pulse by providing short LOW pulse first
+    _delay_us(5);
```

Comment:
is this the default implementation from the functions stuff? I don't think that loop is properly calibrated so you may need to modify it.